src/stapi_fastapi/routers/root_router.py

```diff
@@ -204,42 +242,53 @@ async def get_order(self: Self, order_id: str, request: Request) -> Order:
                 raise AssertionError("Expected code to be unreachable")
 
     async def get_order_statuses(
-        self: Self, order_id: str, request: Request
+        self: Self,
+        order_id: str,
+        request: Request,
+        next: str | None = None,
+        limit: int = 10,
     ) -> OrderStatuses:
-        match await self.backend.get_order_statuses(order_id, request):
-            case Success(statuses):
-                return OrderStatuses(
-                    statuses=statuses,
-                    links=[
-                        Link(
-                            href=str(
-                                request.url_for(
-                                    f"{self.name}:list-order-statuses",
-                                    order_id=order_id,
-                                )
-                            ),
-                            rel="self",
-                            type=TYPE_JSON,
-                        )
-                    ],
+        links: list[Link] = []
+        match await self.backend.get_order_statuses(order_id, request, next, limit):
+            case Success((statuses, Some(pagination_token))):
+                links.append(
+                    self.order_statuses_link(request, "list-order-statuses", order_id)
+                )
+                links.append(
+                    Link(
```

Comment:
put Link construction in a method. The main reason for doing this everywhere is to reduce the length of the case blocks.